src/libAtomVM/interop.h

```diff
@@ -65,6 +67,7 @@ term interop_map_get_value_default(GlobalContext *glb, term map, term key, term
 
 NO_DISCARD InteropFunctionResult interop_iolist_size(term t, size_t *size);
 NO_DISCARD InteropFunctionResult interop_write_iolist(term t, char *p);
+NO_DISCARD InteropFunctionResult interop_iolist_fold(term t, interop_iolist_fold_fun fold_fun, void *accum);
```

Comment:
Does it work?

In theory this should lead to a linking error since you are not providing the inline function in your header (correct me if I am wrong).
Maybe you should do this in `interop.c`:
```c
InteropFunctionResult interop_iolist_fold(term t, interop_iolist_fold_fun fold_fun, void *accum)
{
    return inline_iolist_fold(t, fold_fun, accum);
}
```

So the inline version should be kept private in `interop.c`, for providing better performances for `interop_write_iolist` and `interop_iolist_size`.

CC @pguyot for a second opinion on this topic.

Reply:
My understanding is this is valid and `inline` keyword is only for the current translation unit.

`interop_iolist_fold` is exported and calls to it from `interop_iolist_size` and `interop_write_iolist` are inlined.
Effectively the `interop_iolist_fold` code is there three times in `interop.c.o`.

The question is do we want to export it or could it be static?

Also this welcome factorization definitely conflicts with #735 which will need to be reworked.

Reply:
> My understanding is this is valid and inline keyword is only for the current translation unit.

I did a double check and this code is 100% right.

> The question is do we want to export it or could it be static?

The point of this change is exporting it, so PRs like #689 can be implemented, so yes, we want to export it.